src/Microsoft.ML.Data/Transforms/ConversionsExtensionsCatalog.cs

```diff
@@ -259,6 +337,31 @@ public static KeyToValueMappingEstimator MapKeyToValue(this TransformsCatalog.Co
         /// ]]></format>
         /// </example>
         public static ValueMappingEstimator MapValue(
+            this TransformsCatalog.ConversionTransforms catalog,
+            IDataView lookupMap, string keyColumnName, string valueColumnName, string outputColumnName, string inputColumnName = null)
```

Comment:
>keyColumnName, string valueColumnName [](start = 40, length = 37)

Given that you are providing an actual real honest-to-god dataview as an argument, and not merely the promise for one, the `keyColumnName` and `valueColumnName` argument should almost certainly be a `DataViewSchema.Column` over that data view.

Reply:
Note that it is fine for us to internally just pass along the name in our internal infrastructure, but our external infrastructure should, if we can, allow checking of column (which in this case we totally can, since we have the data view *right here*!).

---
In reply to: [265996171](https://github.com/dotnet/machinelearning/pull/2959#discussion_r265996171) [](ancestors = 265996171)

Reply:
As I mentioned earlier, this is not an API that I am building, it is what we had in our public surface so far. I am happy to improve it if possible however.

We currently already check that the specified `columNames` match existing columns in the `IDataView`. Could you explain why using a `DataViewSchema.Column` would make this simpler from a user stand point?

---
In reply to: [265996567](https://github.com/dotnet/machinelearning/pull/2959#discussion_r265996567) [](ancestors = 265996567,265996171)

Reply:
Given that a user already has an IDataView, how would they call this function? I guess
```
MapValue("Out", lookupMap, lookupMap.Schema["Key"], lookupMap.Schema["Value"], ...)
```
is  safer than
```
MapValue("Out", lookupMap, "Key", "Value", ...)
```

---
In reply to: [266647551](https://github.com/dotnet/machinelearning/pull/2959#discussion_r266647551) [](ancestors = 266647551,265996567,265996171)

Reply:
Ultimately yes, we have *when possible* been trying to shift to that style of API (see, e.g., the work mostly @sfilipi and possibly some others have been doing around shifting the `IDataView` APIs themselves).

Reply:
Ok sounds good, I am changing it to `DataViewSchema.Column`.

---
In reply to: [266695209](https://github.com/dotnet/machinelearning/pull/2959#discussion_r266695209) [](ancestors = 266695209)